frontend/scripts/react-components/profiles/profile-title.component.jsx

```diff
@@ -0,0 +1,56 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import snakeCase from 'lodash/snakeCase';
+import _template from 'lodash/template';
+
+function ProfileTitle(props) {
+  const { template, summary, year, commodityName } = props;
+  function substitute(part) {
+    const interpolate = /{{([\s\S]+?)}}/g;
+    const params = {
+      ...summary,
+      year,
+      commodityName,
+      companyName: summary.nodeName
+    };
+    const snakeCasedParams = Object.entries(params).reduce((acc, [key, value]) => ({
+      ...acc,
+      [snakeCase(key)]: value
+    }));
+    const compiled = _template(part, { interpolate });
+    return compiled(snakeCasedParams);
+  }
+
+  if (!template) {
+    return null;
+  }
+
+  return template.split(' ').map((part, i) => {
+    const hasMustache = /^.*{{.+}}.*$/.test(part);
```

Comment:
Maybe it will fail in cases where there is no space betweeen the mustached part and other ones. Or you have two strings to substitute together Eg. {{quantity}}{{unit}}
Shouldn't it be better to get everything inside the {{}} with a () in the regex?

Reply:
I'm using the regex only to see if the word (previously extracted using `.split(' ')`) has a mustache in it. The part of substituting is very hard so I'm just relying on lodash.

Reply:
Yes, sorry I didn't explain myself. I was thinking in the case that split won't work because we have two moustaches together. Not sure if it will happen